src/encounter.c

```diff
@@ -54,982 +55,951 @@
+    dto->trainerIDs[BATTLER_ENEMY_SLOT_2] = 0;
+    dto->trainerIDs[BATTLER_PLAYER_SLOT_2] = 0;
 
-    TrainerData_Encounter(v1, fieldSystem->saveData, param2);
-    GameRecords_IncrementRecordValue(SaveData_GetGameRecordsPtr(fieldSystem->saveData), RECORD_UNK_008);
-    sub_02050C4C(taskMan, v1, EncEffects_CutInEffect(v1), EncEffects_BGM(v1), param3);
+    TrainerData_Encounter(dto, fieldSystem->saveData, heapID);
+    GameRecords_IncrementRecordValue(SaveData_GetGameRecordsPtr(fieldSystem->saveData), RECORD_TRAINER_BATTLES_FOUGHT);
+    StartEncounter(task, dto, EncEffects_CutInEffect(dto), EncEffects_BGM(dto), resultMaskPtr);
 }
 
-static BOOL sub_020514E8(FieldTask *taskMan)
+static BOOL FieldTask_CatchingTutorialEncounter(FieldTask *task)
 {
-    UnkStruct_02050ACC *v0 = FieldTask_GetEnv(taskMan);
-    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(taskMan);
-    int *v2 = FieldTask_GetState(taskMan);
+    Encounter *encounter = FieldTask_GetEnv(task);
+    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(task);
+    int *state = FieldTask_GetState(task);
 
-    switch (*v2) {
+    switch (*state) {
     case 0:
         MapObjectMan_PauseAllMovement(fieldSystem->mapObjMan);
-        sub_020557DC(taskMan, v0->unk_04, v0->unk_08);
-        (*v2)++;
+        FieldTransition_StartEncounterEffect(task, encounter->introEffectID, encounter->battleBGM);
+        (*state)++;
         break;
+
     case 1:
-        FieldTask_FinishFieldMap(taskMan);
-        (*v2)++;
+        FieldTransition_FinishMap(task);
+        (*state)++;
         break;
+
     case 2:
-        sub_02050ABC(taskMan, v0->unk_10);
-        (*v2)++;
+        CallBattleTask(task, encounter->dto);
+        (*state)++;
         break;
+
     case 3:
-        (*v2)++;
+        (*state)++;
         break;
+
     case 4:
-        FieldTask_StartFieldMap(taskMan);
-        (*v2)++;
+        FieldTransition_StartMap(task);
+        (*state)++;
         break;
+
     case 5:
         MapObjectMan_UnpauseAllMovement(fieldSystem->mapObjMan);
-        sub_020558F0(taskMan);
-        (*v2)++;
+        FieldTransition_FadeIn(task);
+        (*state)++;
         break;
+
     case 6:
-        sub_02050AF0(v0);
-        return 1;
-        break;
+        FreeEncounter(encounter);
+        return TRUE;
     }
 
-    return 0;
+    return FALSE;
 }
 
-void sub_02051590(FieldTask *taskMan)
+void Encounter_NewCatchingTutorial(FieldTask *task)
 {
-    UnkStruct_02050ACC *v0;
-    FieldBattleDTO *v1;
-    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(taskMan);
+    Encounter *encounter;
+    FieldBattleDTO *dto;
+    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(task);
 
-    v1 = FieldBattleDTO_NewCatchingTutorial(11, fieldSystem);
-    v0 = sub_02050ACC(v1, EncEffects_CutInEffect(v1), EncEffects_BGM(v1), NULL);
+    dto = FieldBattleDTO_NewCatchingTutorial(HEAP_ID_FIELDMAP, fieldSystem);
+    encounter = NewEncounter(dto, EncEffects_CutInEffect(dto), EncEffects_BGM(dto), NULL);
```

Comment:
**polish:**
```diff
-    Encounter *encounter;
-    FieldBattleDTO *dto;
-    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(task);
-
-    dto = FieldBattleDTO_NewCatchingTutorial(HEAP_ID_FIELDMAP, fieldSystem);
-    encounter = NewEncounter(dto, EncEffects_CutInEffect(dto), EncEffects_BGM(dto), NULL);
+    FieldSystem *fieldSystem = FieldTask_GetFieldSystem(task);
+    FieldBattleDTO *dto = FieldBattleDTO_NewCatchingTutorial(HEAP_ID_FIELDMAP, fieldSystem);
+    Encounter *encounter = NewEncounter(dto, EncEffects_CutInEffect(dto), EncEffects_BGM(dto), NULL);
```

And so on for the similar functions.

Reply:
This doesn't match for all of these similar functions, unfortunately, which is why I kept them this way for overall consistency.